micro-rdk-nmea/src/parse_helpers/enums.rs

```diff
@@ -0,0 +1,329 @@
+pub trait NmeaEnumeratedField: Sized + From<u32> + ToString {}
+
+/// For generating a lookup data type found in an NMEA message. The first argument is the name of the
+/// enum type that will be generated. Each successive argument is a tuple with
+/// (raw number value, name of enum instance, string representation)
+macro_rules! define_nmea_enum {
+    ( $name:ident, $(($value:expr, $var:ident, $label:expr)),*, $default:ident) => {
+        #[derive(Copy, Clone, Debug)]
+        pub enum $name {
+            $($var),*,
+            $default
+        }
+
+        impl From<u32> for $name {
```

Comment:
Should this be `TryFrom`? Then you wouldn't need the `$default:ident` or associated enum member and could just return an error.

Reply:
So, I had this be the behavior because the gonmea library does not fail on an unrecognized lookup value